meteor/src/lib/MochaRunner.coffee

```diff
@@ -57,36 +51,95 @@ class practical.MochaRunner
       log.return()
 
 
+  runServer: (runId, grep)=>
+    try
+      log.enter("runServer", runId)
+      expect(runId).to.be.a("string")
+      expect(publisher[runId], "publisher").to.be.an("object")
+
+      mochaRunner = new practical.mocha.Mocha()
+      @_addTestsToMochaRunner(mocha.suite, mochaRunner.suite)
+
+      mochaRunner.reporter(practical.mocha.MeteorPublishReporter, {
+        grep: @escapeGrep(grep)
+        publisher: publisher[runId]
+      })
+
+      mochaRunner.run (failures)->
+        log.warn 'failures:', failures
+
+    finally
+      log.return()
+
+
+  _addTestsToMochaRunner: (fromSuite, toSuite)->
```

Comment:
Document it's a recursive function that starts with global suites and adds all sub suites within each global suite